server/api/account.js

```diff
@@ -471,6 +489,16 @@ module.exports.init = function(server) {
             user.password = await hashPassword(userToSet.password, 10);
         }
 
+        if(userToSet.enableGravatar && !user.enableGravatar) {
+            user.emailHash = crypto.createHash('md5').update(user.email).digest('hex');
```

Comment:
Would it be better to do this calculation as a local variable within the `downloadAvatar` method instead of storing it on the user object? We don't need it for anything else and not storing it reduces the risk of it leaking out accidentally in other areas.